sdk/identity/azure-identity/src/main/java/com/azure/identity/VisualStudioCodeCredential.java

```diff
@@ -112,4 +119,27 @@ public Mono<AccessToken> getToken(TokenRequestContext request) {
                 LoggingUtil.logTokenError(LOGGER, identityClient.getIdentityClientOptions(), request, other);
             });
     }
+
+    TokenCredential getBrokerAuthCredential(String clientId) {
+        if (!isVsCodeBrokerAuthAvailable()) {
+            return null;
+        }
+        try {
+            Class<?> builderClass = Class.forName(BROKER_BUILDER_CLASS);
+            Object builder = builderClass.getConstructor().newInstance();
+            AuthenticationRecord authenticationRecord = loadVSCodeAuthRecord();
+            builderClass.getMethod("clientId", String.class).invoke(builder, clientId);
+            builderClass.getMethod("tenantId", String.class).invoke(builder, authenticationRecord.getTenantId());
+            builderClass.getMethod("authenticationRecord", AuthenticationRecord.class)
+                .invoke(builder, authenticationRecord);
```

Comment:
These are all methods that the builder class inherits, instead of making reflective calls here we could cast `builder` to `InteractiveBrowserCredentialBuilder` and call the methods directly.